src/art.rs

```diff
@@ -802,119 +796,128 @@ impl<P: KeyTrait, V: Clone> Node<P, V> {
-        // adjusted and it becomes the Node4's child, while the new Twig node becomes
-        // Node4's inner Twig.
-        if is_prefix_match && cur_node.prefix().len() > key_prefix.len() {
-            let mut n4 = Node::new_node4(key_prefix.into());
-            let mut inner_twig = TwigNode::new(key_prefix.into(), key.clone());
-            inner_twig.insert_mut(value, commit_version, ts);
-            n4.set_inner_twig(inner_twig);
-            let old_node_key = new_key.at(0);
-            let mut old_node = cur_node.clone_node();
-            old_node.set_prefix(new_key);
-            n4.add_child_mut(old_node_key, old_node);
-            return Arc::new(n4);
-        }
-
-        // If the prefixes don't match, create a new Node4 with the old node and a new Twig as children.
+        // Case 1: No prefix match - create new Node4 with split
         if !is_prefix_match {
+            // If the prefixes don't match, create a new Node4 with the old node and a new Twig as children.
             let mut old_node = cur_node.clone_node();
-            old_node.set_prefix(new_key);
-            let mut n4 = Node::new_node4(prefix);
-
-            let k1 = cur_node.prefix().at(longest_common_prefix);
-            let k2 = key_prefix[longest_common_prefix];
+            old_node.set_prefix(new_prefix);
+            let k1 = cur_node.prefix().at(shared_prefix_length);
+            let k2 = key_prefix[shared_prefix_length];
             let new_twig = Node::new_twig(
-                key_prefix[longest_common_prefix..].into(),
+                key_prefix[shared_prefix_length..].into(),
                 key.as_slice().into(),
                 value,
                 commit_version,
                 ts,
             );
-            n4 = n4.add_child(k1, old_node).add_child(k2, new_twig);
 
+            let mut n4 = Node::new_node4(shared_prefix);
+            n4 = n4.add_child(k1, old_node).add_child(k2, new_twig);
             return Arc::new(n4);
         }
 
-        // Continue the insertion process by finding or creating the appropriate child node for the next character.
-        let k = key_prefix[longest_common_prefix];
-        let child_for_key = cur_node.find_child(k);
-        if let Some(child) = child_for_key {
-            let new_child = Node::insert_recurse(
-                child,
-                key,
-                value,
-                commit_version,
-                ts,
-                depth + longest_common_prefix,
-                replace,
-            );
-            let new_node = cur_node.replace_child(k, new_child);
-            return Arc::new(new_node);
-        }
+        // Case 2: Handle prefix match scenarios
+        let cur_prefix_len = cur_node.prefix().len();
+        let key_prefix_len = key_prefix.len();
+
+        match cur_prefix_len.cmp(&key_prefix_len) {
+            // Case 2a: Exact prefix match
+            std::cmp::Ordering::Equal => {
```

Comment:
Let's do `use std::cmp::Ordering` and make it easy on the eyes by using `Ordering::Equal` and the others.

Reply:
Done!